### PR TITLE
Fix build issues related to some examples.

### PR DESCRIPTION
**Pull request by zhanggiene**
This PR contains 

- instructions to download and configure the latest MKL libraries 
- error fix in the Cmake files. 

It will enable users to build examples using the latest MKL libraries. 
With MKL correctly installed and linked, it will solve these #236 #220 #223

### CODE REVIEW

File: examples/logconcave/CMakeLists.txt [hunk 1]
```diff
@@ -8,13 +8,9 @@
...
 
 
-option(DISABLE_NLP_ORACLES "Disable non-linear oracles (used in collocation)" ON)
-option(BUILTIN_EIGEN "Use eigen from ../external" OFF)
-option(USE_MKL "Use MKL library to build eigen" ON)
-
 CMAKE_MINIMUM_REQUIRED(VERSION 2.4.5)
 
-set(MKLROOT $ENV{HOME}/intel/mkl)
+set(MKLROOT /opt/intel/oneapi/mkl/2022.1.0)
```
Comment (papachristoumarios) [inline examples/logconcave/CMakeLists.txt]: should it be mkl/latest instead of the specific version?

File: examples/logconcave/CMakeLists.txt [hunk 2]
```diff
@@ -145,6 +142,8 @@ else ()
   TARGET_LINK_LIBRARIES(simple_sde ${LP_SOLVE} ${BLAS}  "-L${MKLROOT}/lib/intel64 -Wl,--no-as-needed -lmkl_intel_ilp64 -lmkl_gnu_thread -lmkl_core -lgomp -lpthread -lm -ldl")
```
Comment (papachristoumarios) [inline examples/logconcave/CMakeLists.txt]: please create an MKL linking variable as in the tests that contain all of these flags

File: examples/logconcave/README.md [hunk 3]
```diff
@@ -1,4 +1,18 @@
 # Running the Examples
```
Comment (papachristoumarios) [inline examples/logconcave/README.md:1]: Looks good to me

File: examples/logconcave/CMakeLists.txt [hunk 4]
```diff
@@ -8,13 +8,9 @@
 project( VolEsti )
 
 
-option(DISABLE_NLP_ORACLES "Disable non-linear oracles (used in collocation)" ON)
```
Comment (papachristoumarios) [inline examples/logconcave/CMakeLists.txt:11]: why did you remove this options? I think they should be there

**Review by papachristoumarios — CHANGES_REQUESTED** [review]
Thanks for the PR. I have left my comments on specific lines. Also should you fix the other makefiles that depend on MKL (e.g. the test/CMakelists.txt)?

File: examples/logconcave/CMakeLists.txt (hunk 1 above)
Comment (zhanggiene) [reply examples/logconcave/CMakeLists.txt]: Yup,both works but latest is better.

File: examples/logconcave/CMakeLists.txt (hunk 4 above)
Comment (zhanggiene) [reply examples/logconcave/CMakeLists.txt:11]: These are duplicated lines

Comment (papachristoumarios) [reply examples/logconcave/CMakeLists.txt:11]: ok thanks!

**Review by papachristoumarios — COMMENTED** [review]
thanks for the PR! Does it only consider examples and not tests? 
Should you do the same fix on tests since you are running them too (preferably they can be in one PR together)? @vissarion what do you think?

**Review by vissarion — APPROVED** [review]
LGTM, ideally the examples could be added on the CI so that we can see them passing here but this PR is fine without that.
